app.rb

```diff
@@ -0,0 +1,87 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield self[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield self[i], i
+      i += 1
+    end
+  end
```

Comment:
1.Should return an Enumerator if no block is given. For example:
 `p [1, 2, 3].my_each_with_index` should return `#<Enumerator: [1, 2, 3]:my_each_with_index>`

2. When a block is given when self is an array should return the array itself after calling the given block once for each element in self. For example:
`[1,2,3,4,5].my_each_with_index { |num| num < 6}` should return `[1, 2, 3, 4, 5]`

3. When a block is given when self is a range should return the range itself after calling the given block once for each element in self. For example:
`(1..5).my_each_width_index { |num| num < 6}` should return `1..5`

4. When a block is given when self is a range should behave like #each_with_index calls the given block once for each element in self. For example:
`(1..5).my_each_with_index { |num, i| puts "Num: #{num}, index: #{i}"}` should return:
```ruby
Num: 1, index: 0
Num: 2, index: 1
Num: 3, index: 2
Num: 4, index: 3
Num: 5, index: 4
```

5. When a block is given when self is a hash should behave like #each_with_index calls the given block once for each element in self. For example:
```ruby
test = {"a" => 1, "b" => 2, "c" => 3}
test.my_each_with_index {|num, i| puts "#{num}, #{i}" } 
```
should return:
```ruby
["a", 1], 0
["b", 2], 1
["c", 3], 2
```